Camille/src/Enums/Tier.cs

```diff
@@ -1,21 +1,24 @@
-﻿namespace MingweiSamuel.Camille.Enums
-{
-    /// <summary>
-    /// Contains tier names (CHALLENGER, MASTER, etc.)
-    /// </summary>
-    public enum Tier
-    {
-        CHALLENGER,
-        GRANDMASTER,
-        MASTER,
-        DIAMOND,
-        PLATINUM,
-        GOLD,
-        SILVER,
-        BRONZE,
-        IRON,
-
-        /// <summary>In most endpoints, tier will not be provided if related summoner is unranked.</summary>
-        UNRANKED,
-    }
+﻿namespace MingweiSamuel.Camille.Enums
+{
+    /// <summary>
+    /// Contains tier names (CHALLENGER, MASTER, etc.)
+    /// </summary>
+#if USE_SYSTEXTJSON
+  [System.Text.Json.Serialization.JsonConverter(typeof(System.Text.Json.Serialization.JsonStringEnumConverter))]
+#endif
+    public enum Tier
+    {
+        CHALLENGER,
+        GRANDMASTER,
+        MASTER,
+        DIAMOND,
+        PLATINUM,
+        GOLD,
+        SILVER,
+        BRONZE,
+        IRON,
+
+        /// <summary>In most endpoints, tier will not be provided if related summoner is unranked.</summary>
+        UNRANKED,
+    }
```

Comment:
This is from the previous commits, but in the current version `0` or missing values will get parsed as Challenger. API seems to have stopped returning `highestAchievedSeasonTier` so that is one place where that is noticeable.

This will be fixed with proper DTO field nullability which I've already annotated riotapi-schema with. However we should probably start the non-numeric enums from 1 to catch any future occurrences.

Reply:
I am not sure how the returning values work, wouldn't it in this case make sense to put UNRANKED as the `0` in the enum?

Reply:
UNRANKED wouldn't be the same as no value, it seems highestAchievedSeasonTier is straight-up removed https://github.com/RiotGames/developer-relations/issues/94

I'm not worried about it too much, just something I took note of